### PR TITLE
Implement SQLite-backed authentication repository and tests

### PR DESCRIPTION
**Pull request by mohavro**
## Summary
- add a lightweight SQLite-backed database helper to store API credentials seeded from environment variables
- replace the in-memory auth store with a repository that enforces SECRET_KEY presence and expose a protected `/api/users/me` endpoint
- prime the test environment and extend integration coverage to issue and validate tokens against the persistent store

## Testing
- pytest --override-ini addopts=""

------
[Codex Task](https://chatgpt.com/codex/tasks/task_e_690e35c9260083249ff05eea6787802b)

## Summary by Sourcery

Implement a SQLite-backed authentication repository with environment-based seeding of admin credentials, replace the in-memory auth store, add a protected `/api/users/me` endpoint, and extend tests for token issuance and validation against the persistent store.

New Features:
- Introduce a database module with SQLite helpers for connection management, path resolution, and schema initialization
- Add UserRepository for persistent user credential storage with create/update and retrieval operations
- Seed admin credentials from environment variables at startup and enforce SECRET_KEY presence
- Expose a protected `/api/users/me` endpoint that returns the current authenticated user's details
- Extend integration tests to cover token issuance, validation, and access to the protected endpoint

Enhancements:
- Replace bcrypt with pbkdf2_sha256 for password hashing
- Refactor `get_user` and `authenticate_user` to use the repository abstraction instead of an in-memory store

Tests:
- Configure test initialization to reset a SQLite test database and set required environment variables before imports
- Add integration tests for end-to-end authentication flow against the persistent store
- Adjust unit tests in `test_api_main` to support the new database-backed authentication setup

### CODE REVIEW

File: api/auth.py [hunk 1]
```diff
@@ -46,38 +49,131 @@
...
-def authenticate_user(fake_db, username: str, password: str):
+
+def authenticate_user(username: str, password: str, repository: Optional[UserRepository] = None):
     """Authenticate user"""
-    user = get_user(fake_db, username)
+
+    user = get_user(username, repository=repository)
     if not user:
         return False
     if not verify_password(password, user.hashed_password):
```
[the remark this answers is not in the document]
Comment (mohavro) [reply api/auth.py]: @codex review comment and make changes

File: api/auth.py [hunk 2]
```diff
@@ -46,38 +49,193 @@
...
+        disabled=disabled,
+    )
+
+
+_seed_credentials_from_env(user_repository)
+
+if not user_repository.has_users():
+    raise ValueError(
+        "No user credentials available. Provide ADMIN_USERNAME and ADMIN_PASSWORD or pre-populate the database."
+    )
```
[the remark this answers is not in the document]
Comment (mohavro) [reply api/auth.py]: @codex review and fix

[the remark this answers is not in the document]
Comment (mohavro) [reply api/auth.py]: @codex review and fix

**Comment by mohavro** [comment]
@codex review and resolve merge conflict

File: tests/integration/test_api_integration.py [hunk 3]
```diff
@@ -184,6 +186,36 @@
...
+        assert me_response.status_code == 200
+        payload = me_response.json()
+        assert payload["username"] == credentials["username"]
+        assert payload["email"] == os.environ["ADMIN_EMAIL"]
+        assert payload["full_name"] == os.environ["ADMIN_FULL_NAME"]
+        assert payload["disabled"] is False
+
+        invalid_response = client.get("/api/users/me", headers={"Authorization": "Bearer invalid-token"})
+        assert invalid_response.status_code == 401
+
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_api_integration.py]: @codex review comment and make changes. See below:

In tests/integration/test_api_integration.py around lines 189 to 217, the authentication test only asserts the happy path and one invalid-token case; add additional test cases to cover edge conditions: (1) invalid credentials — POST /token with correct username and wrong password and assert 401; (2) disabled user — create or mock a disabled user then attempt token issuance or access and assert the appropriate 401/403 response; (3) missing Authorization header — call GET /api/users/me without the header and assert 401; (4) expired token — if token creation can be mocked or a short-lived token issued, simulate an expired token and assert 401. For each, use the test client fixture, isolate setup/teardown (create/mock users or tokens as needed), and keep assertions minimal: status code and any expected error payload.

File: tests/integration/test_api_integration.py [hunk 4]
```diff
@@ -184,6 +186,36 @@
...
+        assert token_response.status_code == 200
+        token = token_response.json()["access_token"]
+
+        me_response = client.get("/api/users/me", headers={"Authorization": f"Bearer {token}"})
+        assert me_response.status_code == 200
+        payload = me_response.json()
+        assert payload["username"] == credentials["username"]
+        assert payload["email"] == os.environ["ADMIN_EMAIL"]
+        assert payload["full_name"] == os.environ["ADMIN_FULL_NAME"]
+        assert payload["disabled"] is False
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_api_integration.py]: @codex review comment and make changes. See below:

In tests/integration/test_api_integration.py around line 213, the assertion currently hard-codes payload["disabled"] is False; change it to read the expected value from the ADMIN_DISABLED environment variable used by the test setup, convert that env string to the corresponding boolean, and assert payload["disabled"] equals that boolean so the test reflects the seeded ADMIN_DISABLED value rather than a hard-coded False.

**Review by mohavro — COMMENTED** [review]
Ready for merge

**Review by mohavro — COMMENTED** [review]
Review all comments. Make any final changes needed then commit merge.

@codex

**Review by mohavro — COMMENTED** [review]
Commit merge after final comment review and changes.

@codex

**Review by mohavro — COMMENTED** [review]
Merge commit @codex

**Comment by mohavro** [comment]
@codex review all comments, make final changes and commits.

Merge PR
